stdlib/public/Darwin/Accelerate/vImage_Buffer.swift

```diff
@@ -0,0 +1,348 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Accelerate
+
+//===----------------------------------------------------------------------===//
+//
+//  vImage_Buffer
+//
+//===----------------------------------------------------------------------===//
+
+@available(iOS 9999, OSX 9999, tvOS 9999, watchOS 9999, *)
+extension vImage_Buffer {
+    
+    /// Returns the size, in pixels, of a vImage buffer.
+    public var size: CGSize {
+        var mutableSelf = self
+        return vImageBuffer_GetSize(&mutableSelf)
+    }
+    
+    //===----------------------------------------------------------------------===//
+    
+    /// Initializes a vImage buffer of a specified size without memory allocation,
+    /// reporting the preferred alignment.
+    ///
+    /// - Parameter size: The size of the image.
+    /// - Parameter bitsPerPixel: The number of bits in a pixel of image data.
+    /// - Parameter alignment: The preferred alignment is written to this parameter.
+    ///
+    /// - Returns: An initialized vImage buffer.
+    public init?(size: CGSize,
+                 bitsPerPixel: UInt32,
+                 alignment: inout Int) {
+        
+        self.init()
+        
+        if size.width < 0 || size.height < 0 {
+            return nil
+        }
+        let width = vImagePixelCount(size.width)
+        let height = vImagePixelCount(size.height)
+        
+        let error = vImageBuffer_Init(&self,
+                                      height,
+                                      width,
+                                      bitsPerPixel,
+                                      vImage_Flags(kvImageNoAllocate))
+        
+        if error < kvImageNoError {
+            return nil
+        } else {
+            alignment = error
+        }
+    }
+    
+    /// Initializes a vImage buffer of a specified size, reporting any errors.
+    ///
+    /// - Parameter size: The size of the image.
+    /// - Parameter bitsPerPixel: The number of bits in a pixel of image data.
+    /// - Parameter error: Overwritten with the error code if the operation failed.
+    /// - Parameter options: The options to use when performing this operation.
+    ///
+    /// - Returns: An initialized vImage buffer.
+    public init?(size: CGSize,
+                 bitsPerPixel: UInt32,
+                 error: inout Int,
+                 flags options: vImage.Options = .noFlags) {
```

Comment:
The only flags options supported by `vImageBuffer_Init` are `kvImagePrintDiagnosticsToConsole`, `kvImageNoAllocate`, and `kvImageNoFlags`. The first introduces a *third* mode of error handling to this `init` (you already have the error code as well as returning `nil`), and the second is half-baked for reasons described above. This init doesn't need to take a flags parameter, and I don't think you need the inout error param either. Simply return nil for error. That's the idiom for failable inits in Swift.

Reply:
I've made this throw - there are two potential errors, illegal dimensions or a `kvImageMemoryAllocationError`. Other initializers, such as `vImageBuffer_InitWithCGImage` can throw a much wider selection of errors, having all my buffer inits throwable gives a consistent API.

 I'm moving onto removing `flags` now...

Reply:
`flags` removed.